src/stores/RootStore.tsx

```diff
@@ -18,6 +18,8 @@ const RootStore = types
     organisms: types.map(Organism),
     activeAssay: types.maybe(OrganelleRef),
     lockedAssays: types.optional(types.array(OrganelleRef), []),
+    activeSubstance: types.enumeration('SubstanceType', Object.keys(SubstanceType).map(key => SubstanceType[key])),
+    activeSubstanceAmount: types.optional(types.number, 0),
```

Comment:
I think I made things unnecessarily confusing by storing both the type and amount of the addition in a single `Substance` object. Storing them separately like you did here makes updates easier - I just pulled out my old `activeSubstanceManipulation` since it shouldn't be necessary any more.